### PR TITLE
EHN: Fan pcolormesh

### PR DESCRIPTION
**Pull request by mts299**
# Scope 

This removes the double for loop to plot data with `pcolormesh` and fixes the cyclic coloring issue. 

**issue:** #143 

## Approval

**Number of approvals:** 2 approvals - 1 review and 1 tester 
## Test

``` python
import pydarn
from datetime import datetime
import matplotlib.pyplot as plt 

fitacf_file = 'data/20210205.1200.00.cly.a.fitacf'
fitacf_data = pydarn.SuperDARNRead().read_dmap(fitacf_file)
pydarn.Fan.plot_fan(fitacf_data, scan_index=datetime(2021, 2, 5, 12, 5)) 

plt.show()       
```

Figure_1

Note if you run the same code on `develop` you will notice for this fille it has coloring issues with cyclic:
Figure_2

### CODE REVIEW

**Comment by ecbland** [comment]
@mts299 There are some echoes outside the nominal fov in the first plot -- should they also be in the second plot?

**Comment by carleyjmartin** [comment]
> @mts299 There are some echoes outside the nominal fov in the first plot -- should they also be in the second plot?

Dan's older code cut off at 75 range gates when plotting, that's probably the difference here.

**Comment by mts299** [comment]
@ecbland good point need to fix that

**Comment by mts299** [comment]
Done! 

Figure_1

**Comment by carleyjmartin** [comment]
When plotting using: 
```python
file = "/Users/carley/Documents/data/20150305.0601.00.inv.fitacf"
SDarn_read = pydarn.SuperDARNRead(file)
fitacf_data = SDarn_read.read_fitacf()
pydarn.Fan.plot_fan(fitacf_data, scan_index=10, 
            groundscatter=True, colorbar=True, parameter='v', 
            colorbar_label='Velocity', ranges=[0,75])
```

I'm getting the error:
```
Traceback (most recent call last):
  File "dan_test.py", line 26, in <module>
    colorbar_label='Velocity', ranges=[0,75])
  File "/Users/carley/Documents/testing_code/venv-pcolormesh/lib/python3.7/site-packages/pydarn/plotting/fan.py", line 219, in plot_fan
    norm=norm, cmap='Greys')
  File "/Users/carley/Documents/testing_code/venv-pcolormesh/lib/python3.7/site-packages/matplotlib/__init__.py", line 1361, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/Users/carley/Documents/testing_code/venv-pcolormesh/lib/python3.7/site-packages/matplotlib/axes/_axes.py", line 6184, in pcolormesh
    shading=shading, kwargs=kwargs)
  File "/Users/carley/Documents/testing_code/venv-pcolormesh/lib/python3.7/site-packages/matplotlib/axes/_axes.py", line 5699, in _pcolorargs
    C.shape, Nx, Ny, funcname))
TypeError: Dimensions of C (100, 16) are incompatible with X (17) and/or Y (76); see help(pcolormesh)
```

File: pydarn/plotting/fan.py [hunk 1]
```diff
@@ -197,38 +203,26 @@ def plot_fan(cls, dmap_data: List[dict], ax=None,
...
+                      np.ma.masked_array(scan, ~scan.astype(bool)),
+                      norm=norm, cmap=cmap)
+
+        # plot the groundscatter as grey fill
+        if groundscatter:
+            ax.pcolormesh(thetas, rs,
+                          np.ma.masked_array(grndsct,
+                                             ~grndsct.astype(bool)),
+                          norm=norm, cmap='Greys')
+
```
Comment (carleyjmartin) [inline pydarn/plotting/fan.py:220]: Okay so the error is due to setting groundscatter to True, it works fine with it sent to False. So I'm assuming there's something wrong in these lines somewhere causing the above error.

**Comment by carleyjmartin** [comment]
Okay, ignoring the colour differences, I'm unsure if it's just me or if it's just how pcolormesh works. The first below shows the method using pcolormesh, and the second shows the method using the loops. 
It appears that the loop method is creating much cleaner boxes for the data, and the pcolormesh is making some overlap?

pcolormesh:
pcolormesh

loop:
Screen Shot 2021-05-11 at 1 54 11 PM

**Comment by mts299** [comment]
Yea... pcolormesh is known for overlapping its a resolution issue with matplotlib 
this was discussed with summary plots as well

**Comment by ecbland** [comment]
@mts299 You asked me to review this but I'm not sure if you wanted any specific feedback. I'm able to make a fan plot on this branch and it looks beautiful...and I don't see any obvious overlap of the range/beams cells
Figure_1

**Comment by mts299** [comment]
That is all I need  @ecbland just if it looks correct

**Pull request merged**